index.bs

```diff
@@ -2140,7 +2153,27 @@ Base64-encoded string.
       }
 
       browsingContext.CaptureScreenshotParameters = {
-        context: browsingContext.BrowsingContext
+        context: browsingContext.BrowsingContext,
+        ? clip: browsingContext.ClipRectangle
+      }
+
+      browsingContext.ClipRectangle = (
+        browsingContext.BoxClipRectangle //
+        browsingContext.ElementClipRectangle
+      )
+
+      browsingContext.ElementClipRectangle = {
+        type: "element",
+        element: script.SharedReference,
+        ? scrollIntoView: bool
+      }
+
+      browsingContext.BoxClipRectangle = {
+         type: "viewport",
+         x: float,
+         y: float,
+         width: float,
+         height: float
```

Comment:
Out of interest, which scenarios do you think that we can handle with negative width and height?

Reply:
The DOMRect allows the coordinates to be arbitrary floats, so this is matching that. We end up normalizing to positive width/height in the algorithm anyway (basically just subtracting the width from x if it's negative and then replacing it with its absolute value). But also we have to allow fractional values for high dpi screens, and there isn't an easy way to specify non-negative floats, so I think this makes sense.

Reply:
Thanks. That's fair.